File: Min-Cost-Climbing-Stairs.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        dp = [0] * len(cost)
        dp[0], dp[1] = cost[0], cost[1]
        for i in range(2, len(cost)):
            dp[i] = min(dp[i-1] + cost[i], dp[i-2] + cost[i])

        return min(dp[-1], dp[-2])

File: Min-Cost-Climbing-Stairs.go
func minCostClimbingStairs(cost []int) int {
	dp := make([]int, len(cost))
	dp[0] = cost[0]
	dp[1] = cost[1]
	for i := 2; i < len(cost); i++{
		dp[i] = cost[i] + min(dp[i - 1], dp[i - 2])
	}
	return min(dp[len(dp)-1], dp[len(dp)-2])
}